tutorials/shaders/compute_shaders.rst

```diff
@@ -3,70 +3,159 @@
 Using compute shaders
 =====================
 
+This tutorial will walk you through the process of creating a minimal compute
+shader. But first, a bit of background on compute shaders and how they work with
+Godot.
 
-Think of compute shaders as blocks of code that are executed on the GPU for any purpose we want.
-Compute shaders are independent from the graphics pipeline and do not have much fixed-functionality.
-Contrast this with fragment shaders which are used specifically for assigning a color to a fragment in a render target.
-The big benefit of compute shaders over code executed on a CPU is the high amount of parallelization that GPUs provide.
+.. note::
 
-Because compute shaders are independent of the graphics pipeline we don't have any user defined inputs or outputs
-(like a mesh going into the vertex shader or a texture coming out of a fragment shader). Instead, compute shaders
-make changes directly to memory stored on the GPU from which we can read and write using scripts.
+   This tutorial assumes familiarity with non-compute shaders. If you are new to
+   shaders please read :ref:`doc_introduction_to_shaders` and :ref:`your first
+   shader <toc-your-first-shader>` before proceeding with this tutorial.
 
-How they work
--------------
+A compute shader is a special type of shader program that is orientated towards
+general purpose programming. In other words, they are more flexible than vertex
+shaders and fragment shaders as they don't have a fixed purpose (i.e.
+transforming vertices or writing colors to an image). Unlike fragment shaders
+and vertex shaders compute shaders have very little going on behind the scenes,
+the code you write is what the GPU runs and very little else. This can make them
+a very useful tool to offload heavy calculations to the GPU.
```

Comment:
```suggestion
transforming vertices or writing colors to an image). Unlike fragment shaders
and vertex shaders, compute shaders have very little going on behind the scenes. The code you write is what the GPU runs and very little else. This can make them
a very useful tool to offload heavy calculations to the GPU.
```